pkg/core/storage/memory_store.go

```diff
@@ -83,6 +83,28 @@ func (s *MemoryStore) Batch() Batch {
 	}
 }
 
+// Persist flushes all the MemoryStore contents into the (supposedly) persistent
+// store provided via parameter.
+func (s *MemoryStore) Persist(ps Store) (int, error) {
+	s.Lock()
+	defer s.Unlock()
+	batch := ps.Batch()
+	keys := 0
+	for k, v := range s.mem {
+		kb, _ := hex.DecodeString(k)
+		batch.Put(kb, v)
+		keys++
+	}
+	var err error
+	if keys != 0 {
+		err = ps.PutBatch(batch)
+	}
+	if err == nil {
```

Comment:
it will be good to document these lines of code since it's not obvious behavior of flushing memory after received an error from batch.

Reply:
Well, it's the opposite, if we're successful flushing things to `ps` then we can drop the `MemoryStore` contents, if there is a problem --- we can't.